ingestion/src/metadata/ingestion/source/database/vertica/query_parser.py

```diff
@@ -0,0 +1,69 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Vertica usage module
+"""
+from abc import ABC
+from typing import Iterable
+
+from metadata.generated.schema.entity.services.connections.database.verticaConnection import (
+    VerticaConnection,
+)
+from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
+    OpenMetadataConnection,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.tableQuery import TableQuery
+from metadata.ingestion.api.source import InvalidSourceException
+from metadata.ingestion.source.connections import get_connection
+from metadata.ingestion.source.database.query_parser_source import QueryParserSource
+from metadata.ingestion.source.database.vertica.queries import VERTICA_LIST_DATABASES
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class VerticaQueryParserSource(QueryParserSource, ABC):
```

Comment:
Do you know why we are making the class inherit from `ABC`?

Reply:
The `XYZQueryParserSource` does not implement all abstract methods from `QueryParserSource`. It is then used to implement the Usage and Lineage Sources which finish the rest of the implementation